pkg/accumulator/accumulator.go

```diff
@@ -0,0 +1,221 @@
+package accumulator
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"sniffer/pkg/config"
+	"sniffer/pkg/ebpfeng"
+	evData "sniffer/pkg/ebpfev/v1"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+)
+
+const (
+	CACHE_ACCUMULATOR_SIZE = 10
+)
+
+type containersAccumulator struct {
+	accumulatorDataPerContainer map[string]chan evData.EventData
+	registerMutex               sync.Mutex
+}
+
+type CacheAccumulator struct {
+	accumulatorData                 []map[string][]evData.EventData
+	syncReaderWriterAccumulatorData sync.Mutex
+	firstMapKeysOfAccumulatorData   []string
+	cacheAccumulatorSize            int
+	mainDataChannel                 chan *evData.EventData
+	containersData                  containersAccumulator
+	ebpfEngine                      ebpfeng.EbpfEngineClient
+}
+
+type ContainerAccumulator struct {
+	dataChannel chan evData.EventData
+	containerID string
+}
+
+var myContainerID string
+var cacheAccumulator *CacheAccumulator
+
+func CreateAccumulator() *CacheAccumulator {
+	cacheAccumulator = &CacheAccumulator{
+		cacheAccumulatorSize:          CACHE_ACCUMULATOR_SIZE,
+		accumulatorData:               make([]map[string][]evData.EventData, CACHE_ACCUMULATOR_SIZE),
+		firstMapKeysOfAccumulatorData: make([]string, CACHE_ACCUMULATOR_SIZE),
+		mainDataChannel:               make(chan *evData.EventData),
+		containersData: containersAccumulator{
+			accumulatorDataPerContainer: make(map[string]chan evData.EventData),
+		},
+	}
+
+	return cacheAccumulator
+}
+
+func CreateContainerAccumulator(containerID string, dataChannel chan evData.EventData) *ContainerAccumulator {
+	return &ContainerAccumulator{
+		dataChannel: dataChannel,
+		containerID: containerID,
+	}
+}
+
+func (acc *CacheAccumulator) createNewMap(event *evData.EventData, index int) {
+	slice := make([]evData.EventData, 0)
+	m := make(map[string][]evData.EventData)
+	m[event.GetEventContainerID()] = slice
+	acc.accumulatorData[index] = m
+	acc.firstMapKeysOfAccumulatorData[index] = event.GetEventContainerID()
+}
+
+func (acc *CacheAccumulator) findIndexByTimestampWhenAccumulatorDataIsFull(event *evData.EventData) int {
+	index := 0
+	minTimestamp := acc.accumulatorData[0][acc.firstMapKeysOfAccumulatorData[0]][0].GetEventTimestamp()
+	for i := range acc.accumulatorData {
```

Comment:
You could simplify by starting your loop at 1.

```suggestion
	for i := 1; i < len(acc.accumulatorData); i++ {
```

Reply:
I saw a benchmark test that there is no different between those 2 options, there is another reason to this change.
https://pierspowlesland.wordpress.com/2016/08/25/golang-slice-iteration-performance/

Reply:
just so that you don't check in every iteration if i == 0.